### PR TITLE
Toast-Component-New

### PR DESCRIPTION
**Pull request by SakshiGrg**

### CODE REVIEW

File: lib/components/molecules/Toast/Toast.js [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+  return (
+    <>
+      {showToast && (
+        <div
+          className={`${className} ${variation}`}
+          variation={variation}
+          placement={placement}
+          autoDismissTimeout={autoDismissTimeout}
+          autoDismiss={autoDismiss}
```
Comment (vinodloha) [inline lib/components/molecules/Toast/Toast.js]: There is no need for autoDismiss={autoDismiss} to be set here

File: lib/components/molecules/Toast/Toast.js [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+import { ButtonVanilla as Button } from '../../atoms/Button';
+
+const Toast = ({
+  children,
+  autoDismissTimeout,
+  className,
+  isVisible,
+  variation,
+  placement,
+  autoDismiss,
```
Comment (vinodloha) [inline lib/components/molecules/Toast/Toast.js]: autoDismiss Prop looks to be unnecessary and we can make the same decision autoDismissTimeout

Keep autoDismissTimeout defaultValue as 0

File: lib/components/molecules/Toast/Toast.js [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+          variation={variation}
+          placement={placement}
+          autoDismissTimeout={autoDismissTimeout}
+          autoDismiss={autoDismiss}
+        >
+          <Button className="toast-close" onClick={() => close()}>
+            <img src="/images/close.svg" width="12px" alt="close button" />
+          </Button>
+          <div className="toast-text">{children}</div>
+          <span className={`toast-loader toast-dismiss ${variation}`} autoDismiss={autoDismiss} />
```
Comment (vinodloha) [inline lib/components/molecules/Toast/Toast.js]: Make this conditional 

{autoDismissTimeout && <span className={`toast-loader ${variation}`}  />}

toast-dismiss class looks to be unnecessary, make it work with autoDismissTimeout in the CSS

File: lib/components/molecules/Toast/Toast.js [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+    // ... Use hooks
+    /*
+     * This function is dismissing the Toaster after some time interval if autodismiss is true
+     */
+    if (showToast && autoDismissTimeout > 0) {
+      const timer = setTimeout(() => {
+        setshowToast(false);
+      }, autoDismissTimeout);
+
+      clearTimeout(timer);
```
Comment (vinodloha) [inline lib/components/molecules/Toast/Toast.js]: This is a regression.

File: lib/components/molecules/Toast/Toast.js [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+    }
+  });
+
+  /*
+   * This function closes the Toast when close button inside the Toast gets clicked
+   */
+  const close = () => {
+    setshowToast(false);
```
Comment (vinodloha) [inline lib/components/molecules/Toast/Toast.js]: setshowToast --> setShowToast

File: lib/components/molecules/Toast/Toast.js [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+  };
+
+  return (
+    <>
+      {showToast && (
+        <div className={`${className} ${variation}`} variation={variation} placement={placement}>
```
Comment (vinodloha) [inline lib/components/molecules/Toast/Toast.js]: No need of variation={variation} placement={placement}

File: lib/components/molecules/Toast/Toast.js [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+          <Button className="toast-close" onClick={() => close()}>
+            <img src="/images/close.svg" width="12px" alt="close button" />
+          </Button>
+          <div className="toast-text">{children}</div>
+          {autoDismissTimeout ? <span className={`toast-loader ${variation}`} /> : null}
```
Comment (vinodloha) [inline lib/components/molecules/Toast/Toast.js]: Replace this ? : with

{autoDismissTimeout && <span className={`toast-loader ${variation}`} />}

File: lib/components/molecules/Toast/Toast.js [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+        </div>
+      )}
+    </>
+  );
+};
+
+Toast.defaultProps = {
+  isVisible: true,
```
Comment (vinodloha) [inline lib/components/molecules/Toast/Toast.js]: This isVisible: true, is mandatory props so should not be part of defaultProps

File: lib/components/molecules/Toast/Toast.js [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+  autoDismissTimeout: 0,
+  placement: 'top-right',
+  variation: 'default',
```
Comment (vinodloha) [inline lib/components/molecules/Toast/Toast.js]: default --> info

File: lib/components/molecules/Toast/Toast.js [hunk 10]
```diff
@@ -0,0 +1,71 @@
...
+        setShowToast(false);
+      }, autoDismissTimeout);
+    }
+  });
+
+  /*
+   * This function closes the Toast when close button inside the Toast gets clicked
+   */
+  const close = () => {
+    setShowToast(false);
```
Comment (vinodloha) [inline lib/components/molecules/Toast/Toast.js]: use showToast here to change the state

File: lib/components/molecules/Toast/Toast.js [hunk 11]
```diff
@@ -0,0 +1,71 @@
...
+  };
+
+  return (
+    <>
+      {showToast && (
+        <div className={`${className} ${variation}`}>
```
Comment (vinodloha) [inline lib/components/molecules/Toast/Toast.js]: Use classnames package to apply classes.

File: lib/components/molecules/Toast/Toast.js [hunk 12]
```diff
@@ -0,0 +1,71 @@
...
+          <Button className="toast-close" onClick={() => close()}>
+            <img src="/images/close.svg" width="12px" alt="close button" />
+          </Button>
+          <div className="toast-text">{children}</div>
+          {autoDismissTimeout && <span className={`toast-loader ${variation}`} />}
```
Comment (vinodloha) [inline lib/components/molecules/Toast/Toast.js]: ${variation} here is not needed

File: lib/components/molecules/Toast/Toast.js [hunk 13]
```diff
@@ -0,0 +1,71 @@
...
+  const close = () => {
+    setShowToast(false);
+  };
+
+  return (
+    <>
+      {showToast && (
+        <div className={`${className} ${variation}`}>
+          <Button className="toast-close" onClick={() => close()}>
+            <img src="/images/close.svg" width="12px" alt="close button" />
```
Comment (vinodloha) [inline lib/components/molecules/Toast/Toast.js]: Replace this with Icon component

File: lib/components/molecules/Toast/Toast.style.js [hunk 14]
```diff
@@ -0,0 +1,56 @@
+import { css } from 'styled-components';
+
+const ToastStyles = css`
+  position: fixed;
+  display: flex;
+  align-items: center;
+  min-height: 70px;
+  min-width: 300px;
```
Comment (vinodloha) [inline lib/components/molecules/Toast/Toast.style.js]: use --> VW based width as per responsive experience

File: lib/components/molecules/Toast/Toast.style.js [hunk 15]
```diff
@@ -0,0 +1,56 @@
...
+  .toast-text {
+    display: flex;
+    align-items: center;
+    justify-content: space-between;
+    ${props => `${props.theme.typography.FONT_TOAST}`};
+  }
+  .toast-text img {
+    padding-right: 0.8em;
+  }
+  .toast-close {
```
Comment (vinodloha) [inline lib/components/molecules/Toast/Toast.style.js:38]: To be updated along with ICON

File: lib/components/molecules/Toast/Toast.js [hunk 16]
```diff
@@ -0,0 +1,71 @@
...
+   */
+  const close = () => {
+    setShowToast(false);
+  };
+
+  return (
+    <>
+      {showToast && (
+        <div className={`${className} ${variation}`}>
+          <Button className="toast-close" onClick={() => close()}>
```
Comment (vinodloha) [inline lib/components/molecules/Toast/Toast.js]: onClick={() => close()} 

onClick={close}

File: lib/components/molecules/Toast/Toast.js (hunk 13 above)
Comment (SakshiGrg) [reply lib/components/molecules/Toast/Toast.js]: Created a New Issue Task in reference to https://github.com/pagesource/atomic-react-components/issues/190

File: lib/components/molecules/Toast/Toast.style.js (hunk 15 above)
Comment (SakshiGrg) [reply lib/components/molecules/Toast/Toast.style.js:38]: Created a New Issue Task in reference to https://github.com/pagesource/atomic-react-components/issues/190

**Review by vinodloha — APPROVED** [review]

File: lib/components/molecules/Toast/Toast.style.js (hunk 15 above)
Comment (vinodloha) [reply lib/components/molecules/Toast/Toast.style.js:38]: Can you also move your SVG icons lib/static/images to styles/themes/base/icons?

**Pull request merged**